app/controllers/concerns/authentication_and_sso_concerns.rb

```diff
@@ -84,9 +85,9 @@ def set_api_cookie!
 
   # Sets a cookie used by MHV for SSO
   def set_sso_cookie!
-    Rails.logger.info('SSO: ApplicationController#set_sso_cookie!', sso_logging_info)
+    return unless can_set_sso_cookie?
```

Comment:
Is "can" the right word? Might be more readable as simply `set_sso_cookie?`.

Or maybe a slightly over-engineered idea that just popped into my head:

```ruby
def set_sso_cookie!
  return unless can_set_sso_cookie? && should_set_sso_cookie?
  ...
end

def can_set_sso_cookie?
  Settings.sso.cookie_enabled && @session_object.present?
end

def should_set_sso_cookie?
  !@current_user&.has_va_sso
end
```